Sources/SwiftFusion/FactorGraph/FactorGraph.swift

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinGraphs
+
+/// A graph where vertices are parititioned into "left" and "right" sets, and where edges are
+/// always incident to one "left" vertex and one "right" vertex.
+public protocol BipartiteGraph: GraphProtocol
```

Comment:
can we not require that this be a graph? just use it as a graph when you need to.